src/components/EmailNotifications/EmailNotificationsForm.js

```diff
@@ -45,209 +46,223 @@ function EmailNotificationsForm({
     }
   }, [])
 
-  const handleOnSubscribeToNotifications = useCallback(() => {
-    onSubscribeToNotifications(emailAddress)
-  }, [emailAddress, onSubscribeToNotifications])
+  const handleOnSubscribeToNotifications = useCallback(
+    e => {
+      if (emailInvalid || !termsAccepted) {
```

Comment:
Should we make the buttons disabled instead?

Reply:
that is because of the `<form  onSubmit> ` that i didn't push until i can fix that behaviour mentioned bellow